code/drasil-code/lib/Language/Drasil/Code/Imperative/ReadInput.hs

```diff
@@ -98,12 +98,12 @@ splitAtFirst = splitAtFirst' []
         dropDelim [] s = s
         dropDelim _ [] = error "impossible"
 
--- | Converts a list of 'String's to a Matrix 'Expr' of a given 'Space'.
+-- | Converts a list of 'String's to a Clif 'Expr' of a given 'Space'.
 strListAsExpr :: Space -> [String] -> Expr
-strListAsExpr (Vect t) ss = Matrix [map (strAsExpr t) ss]
```

Comment:
you shouldn't do `Matrix` partially; either try for all cases or none, but not just some!